discretisedfield/field.py

```diff
@@ -2735,14 +2737,10 @@ def _writeovf(self, filename, representation='txt', extend_scalar=False):
         .. seealso:: :py:func:`~discretisedfield.Field.fromfile`
 
         """
-        if self.dim != 1 and self.dim != 3:
-            msg = (f'Cannot write dim={self.dim} field.')
-            raise TypeError(msg)
-
-        if extend_scalar and self.dim == 1:
-            write_dim = 3
-        else:
-            write_dim = self.dim
+        write_dim = 3 if extend_scalar and self.dim == 1 else self.dim
+        valuelabels = (' '.join(f'field_{c}' for c in self.components)
+                       if self.components else 'field_x')
+        valueunits = ' '.join(['None'] * self.dim)
```

Comment:
I would probably move `if representation == ...` here to determine `repr_string`. This would allow building header list at once (less lines of code later).